### PR TITLE
C#: Add diagnostic query indicating low database quality

### PR DESCRIPTION
**Pull request by tamasvajk**

### CODE REVIEW

File: csharp/ql/src/Telemetry/DatabaseQualityDiagnostics.ql [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        "see other CodeQL diagnostics reported on the CodeQL status page for more details of possible causes. "
+        +
+        "Addressing these warnings is advisable to avoid false-positive or missing results. If they cannot be addressed, consider scanning C# "
+        +
+        "using either the `autobuild` or `manual` [build modes](https://docs.github.com/en/code-security/code-scanning/creating-an-advanced-setup-for-code-scanning/codeql-code-scanning-for-compiled-languages#comparison-of-the-build-modes)."
+  }
+}
+
+query predicate diagnosticAttributes(DbQualityDiagnostic e, string key, string value) {
+  e = e and // Quieten warning about unconstrained 'e'
```
Comment (hvitved) [inline csharp/ql/src/Telemetry/DatabaseQualityDiagnostics.ql:37]: We usually write `exists(e)` instead.

**Review by hvitved — CHANGES_REQUESTED** [review]
One minor thing, otherwise LGTM. Does this align with how Java reports low quality DBs?

**Comment by tamasvajk** [comment]
> One minor thing, otherwise LGTM. Does this align with how Java reports low quality DBs?

Yes, I took most of the changes from the corresponding java PR: https://github.com/github/codeql/pull/16810.

**Review by hvitved — APPROVED** [review]

**Review by coadaflorin — APPROVED** [review]
LGTM!

**Pull request merged**